parmed/modeller/residue.py

```diff
@@ -128,6 +128,51 @@ def add_atom(self, atom):
         self.atoms.append(atom)
         self._map[atom.name] = atom
 
+    def delete_atom(self, atom):
+        """ Delete an atom from this residue template, along with corresponding bonds.
+
+        Parameters
+        ----------
+        atom : :class:`Atom` or str
+            The atom or atom name to be deleted
+
+        """
+        if type(atom) is str:
+            atom_name = atom
+        else:
+            atom_name = atom.name
+
+        if atom_name not in self._map:
+            raise RuntimeError("Could not find atom '%s' in ResidueTemplate, which contains atoms: %s" % (atom_name, list(self._map.keys())))
+
+        atom = self._map[atom_name]
+
+        # Adjust head and tail if needed
+        if self.head == atom:
+            self.head = None
+        if self.tail == atom:
+            self.tail = None
+
+        # Remove all bonds involving this atom
+        for bond in list(self.bonds):
```

Comment:
why using "list(...)" here?

Reply:
Creating a `list` to iterate over ensures that when I delete one of the elements of `self.bonds`, I don't mess up the iteration if `self.bonds` is a generator.

Reply:
> Creating a list to iterate over ensures that when I delete one of the elements of self.bonds, I don't mess up the iteration if self.bonds is a generator.

Even if it's not a generator you need `list`.  Python gets grumpy if you change the size of the iterator you're iterating over while you're iterating over it.  By casting to a `list`, you avoid this issue.

(sorry for the zombie comment -- it was something I had meant to say a couple weeks ago and think it may still be marginally helpful)